packages/core/test/specHelper.ts

```diff
@@ -178,11 +184,28 @@ class StubXhr {
   }
 
   addEventListener(name: string, callback: () => void) {
-    this.fakeEventTarget.addEventListener(name, callback)
+    if (!this.listeners[name]) {
+      this.listeners[name] = []
+    }
```

Comment:
Did you need a custom event emitter implementation? We are using DOM-based event emitter in a few places because it is good enough for our usecase. If we have more particular requirements, I think we should implement a reusable event emitter class.

Reply:
I noticed with the DOM-based emitter (const eventEmitter = document.createElement('div')) that the "this" keyword is bound to the DIV element. In my case, I want it to be bound to StubXHR. In addition, DIVElement does not expose subscriptions, so we can't make any assertion about them. But you're right, I'm gonna isolate the implementation to make it reusable.